x-pack/packages/kbn-elastic-assistant/impl/knowledge_base/knowledge_base_settings_management/index.tsx

```diff
@@ -0,0 +1,257 @@
+          return !isEsqlSystemEntry(entry);
+        },
+        onEditActionClicked: ({ id }: KnowledgeBaseEntryResponse) => {
+          const entry = entries.data.find((e) => e.id === id);
+          setSelectedEntry(entry);
+          openFlyout();
+        },
+      }),
+    [deleteEntry, entries.data, getColumns, openFlyout]
+  );
+
+  const onDocumentClicked = useCallback(() => {
+    setSelectedEntry({ type: DocumentEntryType.value, kbResource: 'user', source: 'user' });
+    openFlyout();
+  }, [openFlyout]);
+
+  const onIndexClicked = useCallback(() => {
+    setSelectedEntry({ type: IndexEntryType.value });
+    openFlyout();
+  }, [openFlyout]);
+
+  const search: EuiSearchBarProps = useMemo(
+    () => ({
+      toolsRight: (
+        <AddEntryButton onDocumentClicked={onDocumentClicked} onIndexClicked={onIndexClicked} />
+      ),
+      box: {
+        incremental: true,
+        placeholder: i18n.SEARCH_PLACEHOLDER,
+      },
+      filters: [],
+    }),
+    [onDocumentClicked, onIndexClicked]
+  );
+
+  const flyoutTitle = useMemo(() => {
+    // @ts-expect-error TS doesn't understand that selectedEntry is a partial
+    if (selectedEntry?.id != null) {
+      return selectedEntry.type === DocumentEntryType.value
+        ? i18n.EDIT_DOCUMENT_FLYOUT_TITLE
+        : i18n.EDIT_INDEX_FLYOUT_TITLE;
+    }
+    return selectedEntry?.type === DocumentEntryType.value
+      ? i18n.NEW_DOCUMENT_FLYOUT_TITLE
+      : i18n.NEW_INDEX_FLYOUT_TITLE;
+  }, [selectedEntry]);
+
+  if (!enableKnowledgeBaseByDefault) {
+    return (
+      <KnowledgeBaseSettings
+        knowledgeBase={knowledgeBase}
+        setUpdatedKnowledgeBaseSettings={setUpdatedKnowledgeBaseSettings}
+      />
+    );
+  }
+
+  return (
+    <>
+      <EuiPanel hasShadow={false} hasBorder paddingSize="l">
+        <EuiText size={'m'}>
+          <FormattedMessage
+            id="xpack.elasticAssistant.assistant.settings.knowledgeBasedSettingManagements.knowledgeBaseDescription"
+            defaultMessage="The AI Assistant uses Elastic's ELSER model to semantically search your data sources and feed that context to an LLM. Import knowledge bases like Runbooks, GitHub issues, and others for more accurate, personalized assistance. {learnMore}."
+            values={{
+              learnMore: (
+                <EuiLink
+                  external
+                  href="https://www.elastic.co/guide/en/security/current/security-assistant.html"
+                  target="_blank"
+                >
+                  {i18n.KNOWLEDGE_BASE_DOCUMENTATION}
+                </EuiLink>
+              ),
+            }}
+          />
+          {!isKbSetup && !isKbLoading && <SetupKnowledgeBaseButton display={'mini'} />}
+        </EuiText>
+        <EuiSpacer size="l" />
+        <EuiInMemoryTable
+          columns={columns}
+          items={entries.data ?? []}
+          search={search}
+          // onChange={onTableChange}
+        />
```

Comment:
This was in the initial draft PR of this UI (mirroring the work from Conversation/Prompt settings), and will need to be replaced with a paginated `EuiTable` in a subsequent PR since we can have a large number of entries to display (and want to use server side filter).